torch/quantization/fx/_equalize.py

```diff
@@ -220,3 +222,245 @@ def node_supports_equalization(node: Node, modules) -> bool:
+        assert(equalization_qconfig_map.get(weight_node.name, None) is not None)
+
+        weight_eq_obs = equalization_qconfig_map.get(weight_node.name, None).weight()
+        assert(isinstance(weight_eq_obs, _WeightEqualizationObserver))
+        return weight_node, weight_eq_obs
+
+    elif weight_node.op == 'call_function':
+        # TODO
+        return None, None
+
+    return None, None
+
+def get_next_input_obs(node: Node, modules: Dict[str, nn.Module]):
+    """ Gets the following input observer. We can use this function for both
+    obtaining the next input equalization observer and the next input
+    quantization observer
+
+    Returns the node containing the input observer, and the input observer
+    """
+    node = node.next
+    while node.op not in ('call_module', 'call_function', 'output'):
+        node = node.next
+    if node.op in ('output', 'call_function'):
+        return None, None
+
+    assert(isinstance(node.target, str))
+    input_obs = modules[node.target]
+    return node, input_obs
+
+def get_next_equalization_scale(node: Node, modules: Dict[str, nn.Module]) -> torch.Tensor:
+    """ If the next next node is an InputEqualizationObserver then we want to
+    return its equalization scale, else we return 1
+
+    This is used in the case where there are two connecting linear layers:
+        linear1 -> LinearOutObs -> InputEqObs -> linear2
+    In this case, the node given is linear1 and we want to locate the InputEqObs.
+    """
+    # TODO: This code seems a little hacky/hard-coded, so any suggestions for
+    # how to improve it would be greatly appreciated!
+    next_node, _ = get_next_input_obs(node, modules)
+    _, next_next_obs = get_next_input_obs(next_node, modules)
+    if isinstance(next_next_obs, _InputEqualizationObserver):
+        return next_next_obs.equalization_scale
+    return torch.tensor(1)
+
+def scale_input_node(node: Node, modules: Dict[str, nn.Module]) -> None:
+    """ Scales the following input quantization observer's min/max values by
+    updating the values with the scaled min/max values calculated by the input
+    equalization observer
+    """
+    assert(isinstance(node.target, str))
+    input_eq_obs = modules[node.target]
+    assert(isinstance(input_eq_obs, _InputEqualizationObserver))
+
+    input_quant_obs_node = node.args[0]
+    assert(isinstance(input_quant_obs_node, Node) and isinstance(input_quant_obs_node.target, str))
+    input_quant_obs = modules[input_quant_obs_node.target]
+
+    if not isinstance(input_quant_obs, ObserverBase):
+        return
+
+    min_input_scaled, max_input_scaled = input_eq_obs.calculate_scaled_minmax()
+    input_quant_obs.min_val = min_input_scaled
+    input_quant_obs.max_val = max_input_scaled
+
+def scale_weight_node(
+    node: Node,
+    modules: Dict[str, nn.Module],
+    equalization_scale: torch.Tensor,
+    next_equalization_scale: torch.Tensor,
```

Comment:
I think `next_equalization_scale` can be an optional input and we can avoid the extra `mul`s below if it is not set.